inspire_dojson/hep/rules/bd9xx.py

```diff
@@ -272,6 +273,10 @@ def references2marc(self, key, value):
 
     journal_title = get_value(reference, 'publication_info.journal_title')
     journal_volume = get_value(reference, 'publication_info.journal_volume')
+    if journal_title and journal_volume:
+        if journal_title[-1] in string.uppercase:
+            journal_volume = journal_title[-1] + journal_volume
+            journal_title = journal_title[:-1]
```

Comment:
This works, but I don't like this asymmetry. Here's an idea: let's distinguish between pre-filters and post-filters, instead of having a single set of filters. With this abstraction, this would become a pre-filter of `hep2marc`.